src/CloudFlare/Api.php

```diff
@@ -47,24 +55,45 @@ class Api
     private $permissions = null;
 
     /**
+     * Default API endpoint
+     */
+    protected $endpoint = self::ENDPOINT_DEFAULT;
+
+    /**
      * Make a new instance of the API client
      * This can be done via providing the email address and api key as seperate parameters
      * or by passing in an already instantiated object from which the details will be extracted
+     *
+     * Signature 1:
+     * @param $arg1 Object client object
+     *
+     * Signature 2:
+     * @param $arg1 string hosting partner key
+     *
+     * Signature 3:
+     * @param $arg1 string user email
+     * @param $arg2 string user auth key
      */
-    public function __construct()
+    public function __construct($arg1, $arg2 =  null, $arg3 = null)
```

Comment:
My argument to that is that the method signature now becomes ambiguous. Formal parameters are supposed to lend some insight as to what the method accepts and also work in conjunction with docblock commenting.

Not withholding, it also breaks reflection on the method. You'll get inconsistent results with what actually needs to be passed, e.g.

```php
    class Foo
	{
		public function __construct() {
			$args = func_get_args();
			var_dump($args);
		}
	}

	$rfxn = new \ReflectionMethod('Foo', '__construct');
	var_dump($rfxn->getNumberOfRequiredParameters());
```

Alternatively, we can drop support for pre-PHP 5.6 and use the variadic notation, i.e. function foo(...$args) { ... }. Ultimately it's your project, so let me know the direction you want to proceed. That's just my two cents on the matter.